subprojects/base-services/src/test/groovy/org/gradle/internal/classloader/ClasspathUtilTest.groovy

```diff
@@ -36,4 +37,27 @@ class ClasspathUtilTest extends Specification {
         !classpath.asURLs.any { it == JAR_URL || it == HTTP_URL }
         classpath.asURLs.any { it == FILE_URL }
     }
+
+    def "getClasspathForResource for jar scheme with resource name"() {
+        given:
+        def jarUrlWithResourceName = new URL('jar:file:/home/duke/duke.jar!/Test.class')
+
+        when:
+        def file = ClasspathUtil.getClasspathForResource(jarUrlWithResourceName, "Test.class")
+
+        then:
+        "/home/duke/duke.jar" == file.getPath()
```

Comment:
I missed that this assertion and the next one need to be platform agnostic.
Saw the following failure on Windows:
```
Condition not satisfied:
  
  "/home/duke/duke.jar" == file.getPath()
                        |  |    |
                        |  |    \home\duke\duke.jar
                        |  \home\duke\duke.jar
                        false
                        3 differences (84% similarity)
                        (/~)home(/~)duke(/~)duke.jar
                        (\\)home(\\)duke(\\)duke.jar
```

Applies to the one below as well.

Reply:
Updated the test cases.  I don't have a Windows, only verified on Mac, sorry for that.